### ISSUE TITLE
Preconfigured Nick Lock Option - No Appended Nicks

### ISSUE THREAD

**Issue by ndoggac** [thelounge/thelounge#4350]
Instead of attempting to reconnect as fast as possible upon internet connection return, with a new appended nick using the math.random approach, have a feature option where you can force that it will only reconnect using the original configured nick.  If connection is denied, then wait a configured amount of time before attempting reconnect again - again only with the preconfigured nick.

My ISP is terrible, and I lose internet every other day for just a minute or two at a time.  Upon reconnect to internet, thelounge connects back up as fast as possible, often with a randomized nick if the original gets denied because it's still in use.  I don't like this as many of my connections earn me rewards (i.e. meaningless site stats) for connection time, but it has to be using my originally configured nick.  The random nick doesn't get me anything.  To fix I have to manually disconnect, open the connection options, correct the appended nick, save, then reconnect.  Even if I restart, the appended nick is now my default option.  It's a huge pain!  I'd rather wait 10-15 minutes for my "nick in use" to clear on the server, and only connect using my preconfigured nick.  Also, my preconfigured nick should not be permanently overwritten so even after restart I'm affected.

**Comment by fnutt** [comment]
When TL auto-reconnects, it will try to use your nick. If it is taken it will use a new nick. However, it will try to regain your old nick as soon as possible when it is free.

**Comment by ndoggac** [comment]
> When TL auto-reconnects, it will try to use your nick. If it is taken it will use a new nick. However, it will try to regain your old nick as soon as possible when it is free.

This is not the behavior I experience.  It's never reconnected using the originally configured nick, and it also overwrites the pre-configured nick.

**Comment by itsjohncs** [comment]
The Lounge will already try to get your original nickname if it sees that the nick become available. But the implementation is somewhat limited.

After it connects with an alt nick it will look for a `quit` message from your old nick. If it sees one, it will immediately reclaim it. So The Lounge isn't seeing your old nick's `quit` message. I don't know exactly why that's the case, and don't have a workaround to give you, but perhaps someone else will.

I don't think a nick lock option is necessary here. If we get dev time to improve this situation, I think improving the mechanism for reclaiming the nick will be good. A couple quick thoughts on how that could be done:

* Poll to see whether the nick is still around on some interval, maybe with a backoff.
* Use `MONITOR` to ask the server alert us when the nick is free, if the server supports it.

**Comment by ndoggac** [comment]
> After it connects with an alt nick it will look for a quit message from your old nick. If it sees one, it will immediately reclaim it. So The Lounge isn't seeing your old nick's quit message. I don't know exactly why that's the case, and don't have a workaround to give you, but perhaps someone else will.

I'm specifically asking for an implemented workaround here.  Every server I connect to exhibits this same behavior.  So either every server is not implementing the quit message properly or TL is not handling a received quit message properly.  If the former does happen, then better logic around the "quit" message isn't going to make a difference and a TL user has no recourse.  The implementation of a "nick lockout" feature with a configurable timeout for reconnect attempt seems a lot simpler approach to me.

**Comment by ndoggac** [comment]
Wondering if this option is being entertained by devs?  I love TL, but this is a very annoying limitation.

**Comment by brunnre8** [comment]
@ndoggac simply needs someone doing the work, can you send a patch?
xPaw outlined the solutions in this thread

**Comment by c10l** [comment]
I don't think a Nick lock option would be ideal either. The Lounge should *never* change a user-set parameter in the server config.

I understand (although don't necessarily agree with) trying randomly-suffixed nicks *when the original has been taken* but please don't touch my settings!

**Comment by bobokun** [comment]
Is it possible to have this feature implemented? It's really annoying when I have to always re-adjust my nickname back to the original one after it pre-appends some numbers to it

**Comment by zDEFz** [comment]
I also experience the behavior that 

1. TL does try to use the set nick, and if its taken, it uses another
2. Once it has its new name and the old is free, it does not re-take the original nick
3. It overwrites the server settings (nick) to whatever iteration it succeeded with

**Comment by ndoggac** [comment]
> @ndoggac simply needs someone doing the work, can you send a patch? xPaw outlined the solutions in this thread

I see no xPaw outlined solutions in this thread.  I am not a capable enough programmer to implement the feature.

**Comment by qaqland** [comment]
Any update here?

**Comment by qaqland** [comment]
When comes to oftc.net, use `Commands` in User preferences instead of `Cerft` could get the same nickname.

```
/msg NickServ identify <passwd> <name>
```

**Comment by xstefanx** [comment]
I'm still running into this issue. My server reboots once a week, and when it does I always end up with one or two IRC servers that TL uses a random username for, which identifying fails on, and then when I disconnect and reconnect it doesn't auto pull the right nick because the settings for my server have been overwritten with the new appended name it was successful joining with. How can I set the nick setting in user preferences on the server and have that not be altered?

**Comment by c10l** [comment]
Yes, it's really annoying and user-hostile. I get that the nick wasn't free, so the client tries again by adding a counter to the end. That's fine, but it's a transient workaround that should only last for that session. The Lounge should *never* overwrite user settings.

**Comment by brunnre8** [comment]
@c10l it's a bug, all that's missing is for someone to fix it.

There's no secret conspiracy going on, there's no hostility involved, it's simply a broken state machine.

Send a patch, really.
I get it that it sucks, but reiterating the same comment in our issue tracker is not helpful.

But do keep this tracker technical please.
It isn't a board to vent, it's for developers to track issues.
If it ain't fast enough for you, help

**Comment by c10l** [comment]
@brunnre8 You're right, I apologise. I didn't intend to come across as offensive but I appreciate I might have. I also didn't mean that the software is hostile on purpose but I stand by the idea that the behaviour itself is.

You mentioned this is simply a broken state machine, but I don't see how that could be the case. The Lounge opens a writable file descriptor to my user settings, makes a change, saves it and closes it.

User settings is how users tell the software how they want the software to behave. The software should never touch that file without direct user interaction. The only time when the software should write to that file is when the user presses the button to save whatever changes they made to their preferences.

**Comment by brunnre8** [comment]
>You mentioned this is simply a broken state machine, but I don't see how that could be the case

It gets a new nick, is thoroughly confused and thinks that you issued a /nick $whatever. That is exactly the "user interaction" that you keep mentioning.
There's a single field for the old nick, it doesn't cut it.

But really, unless you actually look at the code please just refrain from posting.

You made your point known, no one disagrees. Just needs someone to actually do the work

**Comment by Beersteddy** [comment]
As of today the random nick issue still exists

![image](https://github.com/thelounge/thelounge/assets/91010735/3b22650c-bb45-44b2-bed9-f2e4644dc0d8)

**Comment by brunnre8** [comment]
This is getting way to spammy. Locking the discussion.
If you want to work on this, please hop into our IRC chan on libera and we'll happily unlock the issue and have a productive discussion how to best solve this.

The me too comments are useless however and just make the issue unwieldy.
